app/src/main/java/com/tpp/theperiodpurse/ui/setting/SettingScreenNavigation.kt

```diff
@@ -149,59 +154,74 @@ fun SettingsScreen(
                             navigateUp = { navController.navigateUp() },
-                            color = Color.Transparent,
+                            color = appViewModel.colorPalette.HeaderColor1,
+                            appViewModel = appViewModel
                         ),
                         appViewModel,
                     )
                 }
             }
             composable(route = SettingScreenNavigation.BackUpAccount.name) {
-                BackUpAccountScreen(
-                    appbar = SettingAppBar(
-                        currentScreen = currentScreen.name,
-                        canNavigateBack = navController.previousBackStackEntry != null,
-                        navigateUp = { navController.navigateUp() },
-                        color = Color.White,
-                    ),
-                    navController = navController,
-                    signIn = signIn,
-                    signOut = signout,
-                    context = context,
-                )
+                if (appViewModel != null) {
+                    BackUpAccountScreen(
+                        appbar = SettingAppBar(
+                            currentScreen = "Back Up Account",
+                            canNavigateBack = navController.previousBackStackEntry != null,
+                            navigateUp = { navController.navigateUp() },
+                            color = appViewModel.colorPalette.HeaderColor1,
+                            appViewModel = appViewModel
+                        ),
+                        navController = navController,
+                        signIn = signIn,
+                        signOut = signout,
+                        context = context,
+                        appViewModel = appViewModel
+                    )
+                }
             }
             composable(route = SettingScreenNavigation.DeleteAccount.name) {
                 val context = LocalContext.current
-                DeleteAccountScreen(
-                    appBar = SettingAppBar(
-                        currentScreen = currentScreen.name,
-                        canNavigateBack = navController.previousBackStackEntry != null,
-                        navigateUp = { navController.navigateUp() },
-                        color = Color.White,
-                    ),
-                    navController = navController,
-                )
+                if (appViewModel != null) {
+                    DeleteAccountScreen(
+                        appBar = SettingAppBar(
+                            currentScreen = "Delete Account",
+                            canNavigateBack = navController.previousBackStackEntry != null,
+                            navigateUp = { navController.navigateUp() },
+                            color = appViewModel.colorPalette.HeaderColor1,
+                            appViewModel = appViewModel
+                        ),
```

Comment:
If we are passing in appViewModel to SettingAppBar already, there is no need to pass in color